File: android/app/src/main/java/com/dailyplan/app/domain/model/Enums.kt
// Enums.kt
// 双端一致的枚举原始值定义（规格 §2.2）。持久化一律以 String 落库，禁止用整数 ordinal。

package com.dailyplan.app.domain.model

/** 优先级：高 / 中（默认）/ 低 */
enum class Priority(val raw: String) {
    HIGH("high"), MEDIUM("medium"), LOW("low");   // 默认 MEDIUM
    companion object {
        fun from(raw: String?) = entries.firstOrNull { it.raw == raw } ?: MEDIUM
    }
}

/** 待办来源：语音 / 文字 / 模板 / 补生成 */
enum class TaskSource(val raw: String) {
    VOICE("voice"), TEXT("text"), TEMPLATE("template"), MAKEUP("makeup")
}

/** 同步状态预埋（v1.1 消费，M1 不消费） */
enum class SyncState(val raw: String) {
    LOCAL("local"), DIRTY("dirty"), SYNCED("synced")
}

File: ios/DailyPlan/Models/Enums.swift
// Enums.swift
// 双端一致的枚举原始值定义（规格 §2.2）。
// 持久化一律以 String rawValue 落库，禁止用整数 ordinal（规格 §1.4）。

import Foundation

/// 优先级：高 / 中（默认）/ 低
enum Priority: String, Codable, CaseIterable, Identifiable {
    case high = "high"
    case medium = "medium"
    case low = "low"

    var id: String { rawValue }

    /// 默认优先级（规格 §2.1 默认 "medium"）
    static let `default`: Priority = .medium

    /// 用于 UI 展示的中文名
    var displayName: String {
        switch self {
        case .high: return "高"
        case .medium: return "中"
        case .low: return "低"
        }
    }
}

/// 待办来源：语音 / 文字 / 模板 / 补生成
enum TaskSource: String, Codable {
    case voice = "voice"       // 语音（F2）
    case text = "text"         // 文字（F1）
    case template = "template" // 模板（v1.1 F8）
    case makeup = "makeup"     // 补生成（v1.1 F8）
}

/// 同步状态预埋（v1.1 消费，M1 不消费，规格 §1.6）
enum SyncState: String, Codable {
    case local = "local"   // 默认，仅本地
    case dirty = "dirty"   // 已变更待同步
    case synced = "synced" // 已同步
}
